src/util/number.ts

```diff
@@ -268,6 +270,20 @@ export function getPercentWithPrecision(valueList: number[], idx: number, precis
     return seats[idx] / digits;
 }
 
+/**
+ * Solve the floating point adding problem like 0.1 + 0.2 === 0.30000000000000004
+ * See <http://0.30000000000000004.com/>
+ */
+export function addSafe(val0: number, val1: number): number {
+    const maxPrecision = Math.max(getPrecisionSafe(val0), getPrecisionSafe(val1));
```

Comment:
Can we use `ROUND_SUPPORTED_PRECISION_MAX` in the https://github.com/apache/echarts/blob/a46d2e668f98b17bbade2573b18aad4b4fb35530/src/util/number.ts#L164
to avoid infinity loop? That method is much faster.